Backpack-SwiftUI/Calendar/Classes/Core/CalendarTypeContainerFactory.swift

```diff
@@ -49,97 +39,64 @@ struct CalendarTypeContainerFactory<MonthHeader: View, DayAccessoryView: View>:
             validRange: validRange,
             monthScroll: monthScroll
         ) { month in
-            monthHeader(month)
-            CalendarMonthGrid(
-                monthDate: month,
-                calendar: calendar,
-                validRange: validRange,
-                dayCell: returnMakeCellFunction(),
-                emptyLeadingDayCell: emptyLeadingDayCell,
-                emptyTrailingDayCell: emptyTrailingDayCell,
-                dayAccessoryView: dayAccessoryView
-            )
-        }
-    }
-    
-    func returnMakeCellFunction() -> ((Date) -> CalendarSelectableCell) {
-        return { dayDate in
-            CalendarSelectableCell(
-                selectionType: selectionType,
-                calendar: calendar,
-                accessibilityProvider: calendarAccessibilityConfiguration,
-                dayDate: dayDate,
-                onSelection: handleSelection
-            )
-        }
-        
-    }
-    
-    @ViewBuilder func emptyLeadingDayCell(for emptyDayInfo: EmptyCellInfo) -> some View {
-        switch selectionType {
-        case .range(let selection, _):
-            if
-                case .range(let selectionRange) = selection.wrappedValue,
-                let lastDayOfPreviousMonth = calendar.date(byAdding: .init(day: -1), to: emptyDayInfo.month),
-                let firstDayOfCurrentMonth = calendar.date(byAdding: .init(day: 1), to: lastDayOfPreviousMonth),
-                selectionRange.contains(lastDayOfPreviousMonth),
-                selectionRange.contains(firstDayOfCurrentMonth)
-            {
-                Color(.surfaceSubtleColor)
-            } else {
-                // otherwise we occupy the space with a clear view
-                DefaultEmptyCalendarDayCell()
+            switch selectionType {
```

Comment:
the relevant change here (the point of this PR) is that the `CalendarContainer` is now here, wrapping both the single and range selection calendar month containers, instead of the `CalendarContainer` being inside the single and range respectively. This is what makes the change of selection types dynamically not re-render the entire calendar, as the only thing that's now re-rendered is the months themselves, and not the entire scrollview.

For context, the problem was that before when you changed selection types of the calendar, it'd re-render the whole thing as this switch was the entire calendar including the scrollviews, now the only thing that is re rendered is the months's grids, WITHIN the container scrollview, making the changes to this vlaue NOT trigger a scroll 'reset' 👍

Reply:
So essentially: 
now: 
```
Container {
  Range { Grid } or Single { Grid }
}
```
before:
```
Range {
  Container { Grid }
}
Single {
  Container { Grid }
}
```